src/middlewared/middlewared/plugins/usage.py

```diff
@@ -13,27 +13,35 @@
 
 
 class UsageService(Service):
+
+    FAILED_RETRIES = 3
+
     class Config:
         private = True
 
     async def start(self):
-        if (
-            await self.middleware.call('system.general.config')
-        )['usage_collection']:
+        retries = self.FAILED_RETRIES
+        while retries:
+            if not (await self.middleware.call('system.general.config'))['usage_collection']:
+                break
+
             try:
-                gather = await self.middleware.call('usage.gather')
-                async with aiohttp.ClientSession(
-                    raise_for_status=True
-                ) as session:
+                async with aiohttp.ClientSession(raise_for_status=True) as session:
                     await session.post(
                         'https://usage.freenas.org/submit',
-                        data=gather,
-                        headers={"Content-type": "application/json"},
-                        proxy=os.environ.get("http_proxy"),
+                        data=await self.middleware.call('usage.gather'),
+                        headers={'Content-type': 'application/json'},
+                        proxy=os.environ.get('http_proxy'),
                     )
             except Exception as e:
                 # We still want to schedule the next call
-                self.logger.error(e)
+                self.middleware.logger.error(e)
```

Comment:
Why the change of logger?

Reply:
Consistency ? We normally use `middleware.logger.level`, i can update to the old one if you think that's best

Reply:
Where do we use middleware.logger? Thats just wrong and we shouldnt be doing that for services.

Reply:
@sonicaj ping